File: Threshold/Mean Brightness/threshold.cpp
#include <opencv2/core/utility.hpp>
#include "opencv2/imgproc.hpp"
#include "opencv2/imgcodecs.hpp"
#include "opencv2/highgui.hpp"
#include <iostream>
#include <stdio.h>

using namespace cv;


int MeanBrightness( Mat image)
{
  int nRows = image.rows;
  int nCols = image.cols;
  int image_size = nRows * nCols; //number of pixels
  int soma;
  for(int i = 0; i < image.rows; ++i)
  {
    for (int j = 0; j < image.cols; j++)
    {
      soma += image.at<uchar>(i,j); //getting the sum of all pixel values
    }
  }
  int value = soma/image_size; //get the mean of pixel values
  return value;
}


int main(int argc,const char** argv){
  Mat input_image = imread(argv[1], 1);
  Mat gray_image;
  Mat output;
  cvtColor(input_image,gray_image, COLOR_BGR2GRAY);
  int value = MeanBrightness(gray_image);
  threshold( gray_image, output, 100, 255, THRESH_BINARY);
  imshow("Original", input_image);
  imshow("Threshold", output);
  waitKey(0);
  return 0;
}

File: Threshold/Mean Brightness/threshold.py
import cv2
import numpy as np
from matplotlib import pyplot as plt


def MeanThreshold(image):
    pixels = len(image)*len(image[0]) #get the number of pixels
    soma = 0

    for rows in range(len(image)):
        soma += sum(image[rows]) #sum the pixel value over the entire image

    threshold = soma/pixels #the mean of pixel values is set as the threshold
    ret, dst = cv2.threshold(image, threshold,255,cv2.THRESH_BINARY) #applying the global threshold
    cv2.imshow("Threshold", dst)
    cv2.imshow("Original: ", image)
    cv2.waitKey(0)
    cv2.destroyAllWindows()
    return 0


image = input("Enter the image: ")
img = cv2.imread(image, 0)

MeanThreshold(img)


